rllib/examples/learners/maml_lr_supervised_learning.py

```diff
@@ -0,0 +1,423 @@
+script by `--pause-plot-secs`.
+
+Results to expect
+-----------------
+You should expect to see sometimes alternating test losses ("Total Loss") due to new
+(unseen) tasks during meta learning. In few-shot learning after the meta-learning the
+(few shot) loss should decrease almost monotonically. In the plot you can expect to see
+a decent adaption to the new task after fine-tuning updates of the `RLModule` weights.
+
+With `--iters=70_000`, `--train-batch-size=5`, `--fine-tune-batch-size=5`,
+`--fine-tune-lr=0.01`, `--fine-tune-iters=10`, `--lr=0.001`, `--noise-std=0.0`,
+and no seed defined.
+-------------------------
+
+Iteration: 68000
+Total loss: 0.013758559711277485
+-------------------------
+
+Iteration: 69000
+Total loss: 0.7246640920639038
+-------------------------
+
+Iteration: 70000
+Total loss: 3.091259002685547
+
+Few shot loss: 2.754437208175659
+Few shot loss: 2.7399725914001465
+Few shot loss: 2.499554395675659
+Few shot loss: 2.1763901710510254
+Few shot loss: 1.793503999710083
+Few shot loss: 1.4362313747406006
+Few shot loss: 1.083552598953247
+Few shot loss: 0.7845061421394348
+Few shot loss: 0.5579453110694885
+Few shot loss: 0.4087105393409729
+"""
+import gymnasium as gym
+import matplotlib.pyplot as plt
+import numpy as np
+
+from ray.rllib.algorithms.algorithm_config import DifferentiableAlgorithmConfig
+from ray.rllib.core import DEFAULT_MODULE_ID
+from ray.rllib.core.columns import Columns
+from ray.rllib.core.learner.differentiable_learner_config import (
+    DifferentiableLearnerConfig,
+)
+from ray.rllib.core.learner.training_data import TrainingData
+from ray.rllib.core.rl_module.multi_rl_module import MultiRLModuleSpec
+from ray.rllib.core.rl_module.rl_module import RLModuleSpec
+from ray.rllib.examples.learners.classes.maml_lr_differentiable_learner import (
+    MAMLTorchDifferentiableLearner,
+)
+from ray.rllib.examples.learners.classes.maml_lr_meta_learner import (
+    MAMLTorchMetaLearner,
+)
+from ray.rllib.examples.learners.classes.maml_lr_differentiable_rlm import (
+    DifferentiableTorchRLModule,
+)
+from ray.rllib.policy.sample_batch import MultiAgentBatch, SampleBatch
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.test_utils import add_rllib_example_script_args
+
+# Import torch.
+torch, _ = try_import_torch()
+
+# Define arguments.
+parser = add_rllib_example_script_args(
+    default_iters=70_000,
+)
+
+parser.add_argument(
+    "--train-batch-size",
+    type=int,
+    default=5,
+    help="The number of samples per train and test update (meta-learning).",
+)
+parser.add_argument(
+    "--lr",
+    type=float,
+    default=0.001,
+    help="The learning rate to be used for meta learning (in the `MetaLearner`).",
+)
+parser.add_argument(
+    "--fine-tune-batch-size",
+    type=int,
+    default=10,
+    help="The number of samples for the fine-tuning updates.",
+)
+parser.add_argument(
+    "--noise-std",
+    type=float,
+    default=0.0,
```

Comment:
Should the default be 0.0 here? Or 1.0?

Reply:
Should be `0.0`. We do not want to make it harder than it is already (note: we are basically learning to adapt to infinitely many tasks in few-shot).